lib/src/components/data_table/sort.dart

```diff
@@ -0,0 +1,69 @@
+import 'package:angular2/angular2.dart';
+import 'package:skawa_components/src/components/data_table/data_table_column.dart';
+
+enum SortDirection { asc, desc }
+
+class SortModel {
+  final List<SortDirection> allowedDirections;
+
+  SortDirection activeSort;
+
+  SortModel(this.allowedDirections);
+
+  void toggleSort() {
+    if ((allowedDirections ?? const []).isEmpty) {
+      throw new ArgumentError('SortModel does not have any allowed sort directions');
+    }
+    if (activeSort == null) {
+      activeSort = allowedDirections.first;
+    } else {
+      int directionIndex = allowedDirections.indexOf(activeSort);
+      if (directionIndex == allowedDirections.length - 1) {
+        activeSort = null;
+      } else {
+        activeSort = allowedDirections[directionIndex + 1];
+      }
+    }
+  }
+
+  bool get isAscending => activeSort == SortDirection.asc;
+
+  bool get isDescending => activeSort == SortDirection.desc;
+
+  bool get isSorted => activeSort != null;
+}
+
+@Directive(selector: '[sort]')
+class SkawaDataTableSortDirective {
+  final SkawaDataTableColComponent column;
+
+  SkawaDataTableSortDirective(this.column);
+
+  @Input('sort')
+  set allowedSorts(String allowedSorts) {
+    List<String> directions;
+    if ((allowedSorts ?? '').isNotEmpty) {
+      directions = allowedSorts.split(',').map((s) => s.trim()).toList(growable: false);
+    } else {
+      directions = directionMap.keys.toList(growable: false);
+    }
+    if (directions.isEmpty || directions.any((s) => directionMap[s] == null)) {
+      throw new ArgumentError(
+          'SkawaDataTableSortDirective accepts only "asc" and/or "desc" as sort directions. Use comma separated values for both directions.');
+    }
+    column.sort = new SortModel(directions.map((s) => directionMap[s]).toList(growable: false));
+  }
+
+  @Input()
+  set defaultSort(String sort) {

```

Comment:
there are no way to preserve multiply column to have `defaultSort`

Reply:
`defaultSort` is to mark a column for initial sorting in the table. There can be only one sorted column at a given time in the table, so this is intentional. However the naming may be confusing a bit, so I renamed this Input to `initialSort`. For preferred sort direction, you can pass `sortable="asc, desc"` or `sortable="desc, asc"` to the directive.

Reply:
I wanted to write prevent but i writed preserve

Reply:
Fixed